### DISCUSSION TITLE
LVars - Changed or disappear

### DISCUSSION THREAD

**Discussion by Framac23** [flybywiresim/aircraft#6539] [Help]
Hi
In almost all versions the LVars change or disappear.
I'm making some electronic modules and every time I update the FBW the program crashes.
Now some that I use have disappeared and I can't find them.
A32NX_AUTOPILOT_SPEED_SELECTED
A32NX_FCU_SPD_MANAGED_DASHES
A32NX_FCU_SPD_MANAGED_DOT

Is there any method of keeping or knowing the equivalents so I don't keep changing my program?
Thanks

**Comment by aguther** [comment]
I don't know what is happening on your end, but all those variables are still in place.

Besides that, keep in mind that a lot of systems are still in development, and we try our best to be as much as compatible. This isn't an easy task at all because even systems development is challenging.

The alternative is to do hidden development and release the plane when it's "ready".

**Reply by Framac23** [reply]
Hi Aguther.
This happened after installing the latest development version.
I installed the stable version and the lvars are recognized again.
I understand and accept that it is a development version, I was a little frustrated because I would have to redo all the code, but I will stick with the stable version, and that way I have everything working. :)
But it would be nice if there was some information about the lvar for those who are developing modules and external programs.

**Reply by tracernz** [reply]
The L vars are documented here https://github.com/flybywiresim/a32nx/blob/master/docs/a320-simvars.md, and the events here: https://github.com/flybywiresim/a32nx/blob/master/docs/a320-events.md (both for development version)

**Reply by Framac23** [reply]
@aguther and @tracernz I found the problem.
Sometimes my FSUIPC Wasm don't load all LVars at start, I don't know if its a bug but I will post in FSuipc forum.
I have to disable and re-enable it and everything works fine.
But after all the LVars are available, my mistake.
Thanks for your help.

**Reply by aguther** [reply]
Ok. For those reasons it would be great if SimConnect would support it directly. Any other approach is not really reliable.

**Reply by AdamNZ** [reply]
Not sure whether this is related or not - but I often have to "re-activate" FSUIPC during MSFS sessions by switching to the FSUIPC window and messing with the axes assignations (I use profiles). It's as if FSUIPC stops communicating with the sim, for some reason. This happens with any aircraft - not just the A32NX. I may need to check the FSUIPC forums as well.
